server/src/test/java/org/opensearch/cluster/coordination/JoinTaskExecutorTests.java

```diff
@@ -582,6 +582,14 @@ public void testPreventJoinClusterWithRemoteStoreNodeWithPartialAttributesJoinin
             );
             assertTrue(
                 e.getMessage().equals("joining node [" + joiningNode + "] doesn't have the node attribute [" + nodeAttribute.getKey() + "]")
+                    || e.getMessage()
```

Comment:
dont we need changes in Join Executor Validator to ensure all nodes have same remote state repo or that is taken care automatically?

Reply:
There is no change in Join Executor. It will still validate the all the repos (segment, translog, cluster_state) in the joining node are the same as the existing node. Reference: https://github.com/soosinha/OpenSearch/blob/main/server/src/main/java/org/opensearch/cluster/coordination/JoinTaskExecutor.java#L533C25-L552